### PR TITLE
Remove navigation store's embedded property 

### LINKED ISSUE #2159: Remove code for "embedded mode"
## Problem

Openverse used to be embedded in WordPress theme, and had to have a way of setting the `embedded` properties and pass messages to the main window. This is not necessary anymore.

## Description

We should remove:
- `navigation` store's `embedded` property
- ~/utils/send-message

These are set here:
https://github.com/WordPress/openverse/blob/8bd7c4525ce0275b4ee95448c1de6639461624b8/frontend/src/middleware/middleware.ts#L36-L44

### PR DESCRIPTION
**Pull request by YashGaur000**
## Fixes

Fixes #2159 by @YashGaur000 

## Description
The code checks if the query parameter "embedded" exists and has a value of "true". If it does, it sets the "isEmbedded" property in the "navigation store" to true. But we needed to remove this code snippets,

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/middleware/middleware.ts [hunk 1]
```diff
@@ -33,9 +33,6 @@ const middleware: Middleware = async ({
 
   const navigationStore = useNavigationStore($pinia)
 
-  if ("embedded" in query) {
-    navigationStore.setIsEmbedded(query.embedded === "true")
-  }
   if (process.client) {
```
Comment (obulat) [inline frontend/src/middleware/middleware.ts:36]:
```suggestion
  if (process.client) {
```
Could you also remove the `sendWindowMessage`?

**Review by obulat — CHANGES_REQUESTED** [review]
This is a great start, @YashGaur000!

Could you also remove the `sendWindowMessage` call from lines 36-41 in the updated `middleware` file, and the `isEmbedded` property and methods that update it from the `navigation` store: `frontend/src/stores/navigation.ts` ?

**Review by obulat — CHANGES_REQUESTED** [review]
To finish fixing, we need to also remove:
- import on line 1
- `route` param on line 29
- `if(process.client)` statement on lines 36-37

**Review by obulat — CHANGES_REQUESTED** [review]
We can also delete `frontend/src/plugins/url-change.ts`. After that, we would need to delete it from the `plugins` array in `nuxt.config.ts`

**Comment by YashGaur000** [comment]
Why is there a need to delete it? Can you help me with this, I'm new to open source.

**Comment by obulat** [comment]
> Deleted frontend/src/plugins/url-change.ts and from the plugins array in nuxt.config.ts also.
> 
> *Why is there a need to delete it? Can you help me with this, I'm kinda new to open source.

Sure, @YashGaur000! I'm sorry I didn't give more details in the issue. The `isEmbedded` property is used in several places, and you've removed it from them. Previously, Openverse was hosted inside the wordpress.org iframe. Now, the old URL (https://wordpress.org/openverse) is only used for redirect to the new domain, `https://openverse.org`

`frontend/src/plugins/url-change.ts` is the file that was used to sync the URL between the iframe and the Openverse app. It is not needed anymore.

I've removed the unused tests, and now this PR is ready 🚀 Thank you for your contribution, and welcome to Openverse!

**Review by obulat — APPROVED** [review]

**Comment by YashGaur000** [comment]
Thank you for your guidance and for making me understand the these issue needed to be altered. And Thank you for welcoming me in openverse.

**Comment by YashGaur000** [comment]
Can you please tell me why this other review is pending? And will it even be reviewed?

**Comment by YashGaur000** [comment]
Thank you so much to approve this pull request @obulat. Can I know when
this pull request be merged? And how do I become the contributor in
openverse?

**Comment by obulat** [comment]
> Can I know when this pull request be merged?

Our policy is that every PR needs 2 approvals to be merged, @YashGaur000.

**Comment by YashGaur000** [comment]
When will the other one review it? Will she even review it?

**Review by dhruvkb — APPROVED** [review]
LGTM, thanks for the contribution @YashGaur000! Please also note that you're free to work on another issue while a PR is awaiting a second approval.

**Comment by YashGaur000** [comment]
Okay. Thank you to inform me. I appreciate that a lot.

**Pull request merged**